Source/WebKit/UIProcess/API/gtk/DropTargetGtk4.cpp

```diff
@@ -173,8 +212,29 @@ void DropTarget::accept(GdkDrop* drop, std::optional<WebCore::IntPoint> position
             } else if (mimeType == "text/uri-list"_s) {
                 gsize length;
                 const auto* uriListData = g_bytes_get_data(data.get(), &length);
-                if (length)
-                    m_selectionData->setURIList(String::fromUTF8(reinterpret_cast<const char*>(uriListData), length));
+                if (length) {
+                    String uriListString(String::fromUTF8(reinterpret_cast<const char*>(uriListData), length));
+                    for (auto& line : uriListString.split('\n')) {
+                        line = line.trim(deprecatedIsSpaceOrNewline);
+                        if (line.isEmpty())
+                            continue;
+                        if (line[0] == '#')
+                            continue;
+
+                        // If we have file transfers from the portal, ignore file:// URIs.
+                        URL url { line };
+                        if (transferredFilesFromPortal && url.isValid()) {
+                            GUniqueOutPtr<GError> error;
+                            GUniquePtr<gchar> filename(g_filename_from_uri(line.utf8().data(), 0, &error.outPtr()));
+                            if (!error && filename)
+                                continue;
+                        }
+
+                        if (!m_uriListBuilder.isEmpty())
+                            m_uriListBuilder.append("\r\n");
+                        m_uriListBuilder.append(line);
+                    }
```

Comment:
I'm not sure if this kind of modification / sanitization is allowed

Reply:
I'm not sure I understand what you are doing here.

Reply:
This block of code basically filters out `file://`s from the URI list.

The URI list comes as a continuous string separated by `\n`. So I break the URI string by newline; filter out `file://` URIs; and rebuild the (continuous) list in `m_uriListBuilder`.

After both mimetypes and the GdkFileList are processed, I merge the `GFile`s as `file://` URIs into the URI list.